crates/circuit/src/circuit_data.rs

```diff
@@ -1167,4 +1167,9 @@ impl CircuitData {
             py_op: RefCell::new(inst.py_op.clone()),
         })
     }
+
+    /// Rust native method, returns an iterator over all the instructions present in the circuit.
```

Comment:
We won't need to say "Rust native method" - _all_ the methods we call from Rust space are Rust native, even the ones that PyO3 provides a wrapper for so that Python can call them.  Rust doesn't pay any overhead to call `pymethod`s.